### PR TITLE
Created state machine and tests and added new lines to readme

### PR DESCRIPTION
**Pull request by ester-agishtein**

### CODE REVIEW

File: src/main/java/touro/snake/SnakeHeadStateMachine.java [hunk 1]
```diff
@@ -26,8 +26,30 @@ public Direction getDirection() {
...
+    }
+    public boolean checkOrientation(Direction dir){
+        boolean vertical = false;
+        if (dir == Direction.North || dir == Direction.South){
+            vertical = true;
+        }
+        if (dir == Direction.East || dir == Direction.West){
+            vertical = false;
+        }
+        return vertical;
```
Comment (shweinfeld) [inline src/main/java/touro/snake/SnakeHeadStateMachine.java:52]: This was a great way to implement the method.

File: src/main/java/touro/snake/Direction.java [hunk 2]
```diff
@@ -5,4 +5,5 @@
  */
 public enum Direction {
     North, East, South, West
+
```
Comment (andrewoid) [inline src/main/java/touro/snake/Direction.java:8]: remove extra space

File: src/test/java/touro/snake/SnakeHeadStateMachineTest.java [hunk 3]
```diff
@@ -8,62 +8,154 @@
 
     @Test
     public void turnTo_eastToSouth() {
-        throw new UnsupportedOperationException("Not Implemented Yet.");
+        Direction direction = Direction.East;
```
Comment (andrewoid) [inline src/test/java/touro/snake/SnakeHeadStateMachineTest.java:11]: Please separate into given-when-then

File: src/test/java/touro/snake/SnakeHeadStateMachineTest.java [hunk 4]
```diff
@@ -8,62 +8,154 @@
...
+        assertEquals(Direction.West, result);
     }
 
+    @Test
+    public void checkOrientation_vertical(){
+        Direction direction = Direction.North;
+        Direction newDirection = Direction.South;
+        SnakeHeadStateMachine machine = new SnakeHeadStateMachine(direction);
+        boolean isVertical = machine.checkOrientation(newDirection);
+        assertEquals(true,isVertical);
```
Comment (andrewoid) [inline src/test/java/touro/snake/SnakeHeadStateMachineTest.java:135]: assertTrue

File: src/test/java/touro/snake/SnakeHeadStateMachineTest.java [hunk 5]
```diff
@@ -8,62 +8,154 @@
...
+    }
+    @Test
+    public void checkOrientation_horizontal(){
+        Direction direction = Direction.East;
+        Direction newDirection = Direction.West;
+        SnakeHeadStateMachine machine = new SnakeHeadStateMachine(direction);
+        boolean isVertical = machine.checkOrientation(newDirection);
+        assertEquals(false,isVertical);
```
Comment (andrewoid) [inline src/test/java/touro/snake/SnakeHeadStateMachineTest.java:143]: assertFalse

File: src/main/java/touro/snake/SnakeHeadStateMachine.java [hunk 6]
```diff
@@ -26,8 +26,30 @@ public Direction getDirection() {
      * @param newDirection
      */
     public void turnTo(Direction newDirection) {
-        // Agishtein
-        throw new UnsupportedOperationException("Not Implemented Yet.");
+        boolean dirVert = this.checkOrientation(direction);
+        boolean newDirVert = this.checkOrientation(newDirection);
+
+        if(dirVert && newDirVert){
```
Comment (andrewoid) [inline src/main/java/touro/snake/SnakeHeadStateMachine.java:32]:
```
if ((dirVert && newDirVert) || (!dirVert && !newDirVert)) {
   return;
}
```

File: src/main/java/touro/snake/SnakeHeadStateMachine.java [hunk 7]
```diff
@@ -26,8 +26,30 @@ public Direction getDirection() {
...
+            return;
+        }
+        else{
+            direction = newDirection;
+        }
+    }
+    public boolean checkOrientation(Direction dir){
+        boolean vertical = false;
+        if (dir == Direction.North || dir == Direction.South){
+            vertical = true;
```
Comment (andrewoid) [inline src/main/java/touro/snake/SnakeHeadStateMachine.java:47]: return true;

File: src/main/java/touro/snake/SnakeHeadStateMachine.java [hunk 8]
```diff
@@ -26,8 +26,30 @@ public Direction getDirection() {
...
+        }
+        if (dir == Direction.East || dir == Direction.West){
+            vertical = false;
```
Comment (andrewoid) [inline src/main/java/touro/snake/SnakeHeadStateMachine.java:50]: return false;

**Pull request merged**